pkg/client/cli/output/output.go

```diff
@@ -0,0 +1,104 @@
+// Package output provides structured output for *cobra.Command.
+// Writing JSON to stdout is enable by setting the --output=json flag.
+package output
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func WithOutput(ctx context.Context, cmd *cobra.Command) context.Context {
+	next := cmd.PersistentPreRun
+	o := output{
+		stdout: cmd.OutOrStdout(),
+		stderr: cmd.ErrOrStderr(),
+	}
+
+	cmd.PersistentPreRun = func(cmd *cobra.Command, args []string) {
+		// never output help messages as json
+		if cmd.RunE == nil {
+			return
+		}
+
+		cmd.RunE = o.runE(cmd.RunE)
+
+		if next != nil {
+			next(cmd, args)
+		}
+	}
+
+	return context.WithValue(ctx, key{}, &o)
+}
+
+func Outputs(ctx context.Context) (stdout, stderr io.Writer) {
```

Comment:
The move into an `output` package introduces stuttering, `output.Outputs`. Consider changing the `Outputs` to something that would be more suitable when prefixed with `output`. Perhaps `output.Formatted` ?